cmd/telemetrygen/Dockerfile

```diff
@@ -1,10 +1,3 @@
-FROM golang:1.21 AS build
-
-WORKDIR /src
-ADD . /src
-
-RUN CGO_ENABLED=0 go build -o /telemetrygen
-
```

Comment:
I feel like we normally handle the build as part of the dockerfile to make it easier for others to build themselves.  If we remove this won't users depending on this file have to start building the binary separately instead of doing `docker build .`?

The gains could be worth it, but I'd like to discuss the impact of this particular change.

Reply:
The only way to really build this docker image for prod is what happens in the github workflow:
```
      - name: Push telemetrygen to Github packages
        uses: docker/build-push-action@v5
        with:
          context: cmd/telemetrygen
          push: true
          tags: ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:latest
          platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
```

What that allows us to do is to declare one docker image name with multiple architectures, and when a customer pulls, they get the image that matches their arch.

https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/

This is the image we push:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pkgs/container/opentelemetry-collector-contrib%2Ftelemetrygen/170736692?tag=v0.93.0

<img width="745" alt="Screenshot 2024-02-01 at 08 38 43" src="https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/16758/ad53e883-68eb-47ee-b4e0-6ca6008fe9f7">

I think we could offer a README entry to avoid that folks try to run a `docker build -t test .` in this folder and instead run `make docker-telemetrygen` which will do the work safely and give them a multi-arch image.

Reply:
Let's throw the `make docker-telemetry` command into the readme as well.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30989/commits/f65a79cde7c4628251ce8fa07dfdfde720c504b8 done

Reply:
I dont see `make docker-telemetrygen` in there

Reply:
OK I reworked the Makefile instead